### PR TITLE
add typescript definition file

### PR DESCRIPTION
**Pull request by byr-gdp**

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+interface IOptions {
+  messages?: any;
+  className?: any;
+  autoForceUpdate?: boolean;
+  element?: any;
+}
+
+type IRule = 'accepted' | 'after' | 'after_or_equal' | 'alpha' | 'alpha_space' | 'alpha_num' |' alpha_num_space' | 'alpha_num_dash' | 'alpha_num_dash_' | 'array' | 'before' | 'before_or_equal' | 'between' | 'boolean' | 'card_exp' | 'card_num' | 'currency' | 'date' | 'date_equals' | 'email' | 'in' | 'integer' | 'max' | 'min' | 'not_in' | 'not_regex' | 'numeric' | 'phone' | 'regex' | 'required' | 'size' | 'string' | 'typeof' | 'url';
```
Comment (stuyam) [inline src/index.d.ts]: I think `alpha_num_dash_` is meant to be `alpha_num_dash_space`?

**Comment by stuyam** [comment]
Fixes: #94

File: src/index.d.ts (hunk 1 above)
Comment (byr-gdp) [reply src/index.d.ts]: yeah, that's right, my mistake...

Comment (byr-gdp) [reply src/index.d.ts]: I noticed you have update the wrong spell word, so I am not going to open another pr.

**Pull request merged**